src/actions/plugins/cleverF.ts

```diff
@@ -0,0 +1,253 @@
+import * as vscode from 'vscode';
+
+import { configuration } from '../../configuration/configuration';
+import { Mode } from '../../mode/mode';
+import { Notation } from '../..//configuration/notation';
+import { VimState } from '../../state/vimState';
+import { Position } from 'vscode';
+import { RegisterAction } from '../base';
+import { BaseMovement, IMovement, failedMovement } from '../baseMotion';
+import { findHelper, MoveRepeat, MoveRepeatReversed } from '../motion';
+
+/**
+ * State of Clever-f
+ */
+let posF = [-1, -1];
+let beforeCleverFAction: string;
+let previousMode: Mode;
+
+/**
+ * CleverF f command
+ */
+@RegisterAction
+class ActionCleverForwardCommand extends BaseMovement {
+  override modes = [Mode.Normal, Mode.Visual];
+  keys = ['f'];
+
+  // This command is Not Enabled when vim.clever-F is false.
+  public override doesActionApply(vimState: VimState, keysPressed: string[]): boolean {
+    return configuration.cleverF && super.doesActionApply(vimState, keysPressed);
+  }
+
+  public override async execActionWithCount(
+    position: Position,
+    vimState: VimState,
+    count: number
+  ): Promise<Position | IMovement> {
+    count ||= 1;
+    // First if is repeat "f" movement. Ex) "f","a","f" called
+    // Second else if is repeat "F" movement.Ex) "F","a","f" called
+    // Third else is first "f" command. Ex) "f","a" called
```

Comment:
Excuse me, your English is weird. I guess what you want to say like this.

The first if-clause states that the current operator is `f` and also the last operator is `f`.
Next else-if-clause states that the current operator is `f`-command and the last operator is `F`.
The final else-clause states that the current operator is the initial `f`.


Is this true?

Reply:
thank you for add comment!
Sorry for my word is not corrected
that is true. 

I will fix my comment!

Reply:
fixed

Reply:
Thank you. I confirmed that my concern is solved. Thank you for the great extra plugin :)